pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java

```diff
@@ -278,11 +279,42 @@ public static void validateIngestionConfig(TableConfig tableConfig, @Nullable Sc
                 "Arguments of a transform function '" + arguments + "' cannot contain the destination column '"
                     + columnName + "'");
           }
+          columnToFunction.put(columnName, expressionEvaluator);
+        }
+        Map<String, Integer> chainDepth = new HashMap<>();
+        for (String column : columnToFunction.keySet()) {
+          Preconditions.checkState(transformFunctionChainDepth(column, columnToFunction, chainDepth) <= 2,
```

Comment:
I think it might be helpful to log the erroneous function with 2 derived columns that are chained.

Reply:
done